lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/SharedKey.hs

```diff
@@ -102,10 +108,15 @@ constructAddressFromIx pTemplate dTemplate ix =
             CA.unAddress $
             paymentAddress tag
             (paymentCredential pScript')
+        role' = case role of
+            UtxoExternal -> CA.UTxOExternal
+            UtxoInternal -> CA.UTxOInternal
+            MutableAccount ->
+                error "role is specified only for payment credential"
         pScript =
-            replaceCosignersWithVerKeys CA.UTxOExternal pTemplate ix
+            replaceCosignersWithVerKeys role' pTemplate ix
         dScript s =
-            replaceCosignersWithVerKeys CA.Stake s ix
+            replaceCosignersWithVerKeys CA.Stake s minBound
```

Comment:
Ah! No address discovery for the staking part, right. 🤓

Reply:
yes, at this moment we are leaving staking consideration for the future. We will have here several options, one will be to have ix=0, we could also sneak external staking key!!! I hope to discuss it when delegation for multisig ticket will be in progress. Until know ix=0 is fine and complies with the least surprise principle ;-)